MobileWallet/Backup/BackupPrompts.swift

```diff
@@ -0,0 +1,203 @@
+                    ctaButton: NSLocalizedString("wallet_backup.prompt_2.cta_button", comment: "Wallet backup prompts"),
+                    cancelButton: NSLocalizedString("wallet_backup.prompt_2.cancel_button", comment: "Wallet backup prompts")
+                )
+            case .third:
+                return ModalContent(
+                    title: NSLocalizedString("wallet_backup.prompt_3.title", comment: "Wallet backup prompts"),
+                    boldTitle: NSLocalizedString("wallet_backup.prompt_3.title_bold", comment: "Wallet backup prompts"),
+                    description: NSLocalizedString("wallet_backup.prompt_3.description", comment: "Wallet backup prompts"),
+                    ctaButton: NSLocalizedString("wallet_backup.prompt_3.cta_button", comment: "Wallet backup prompts"),
+                    cancelButton: NSLocalizedString("wallet_backup.prompt_3.cancel_button", comment: "Wallet backup prompts")
+                )
+            }
+        }
+    }
+
+    static let shared = BackupPrompts()
+
+    private init () {}
+
+    func check(_ vc: UIViewController) {
+        guard let wallet = TariLib.shared.tariWallet else {
+            return
+        }
+
+        for type in PromptTypes.allCases {
+            //If they have been shown this once, skip over this prompt
+            guard UserDefaults.standard.bool(forKey: type.userDefaultsKey) == false else {
+                continue
+            }
+
+            var incomingTxs = wallet.pendingInboundTransactions.0?.count.0 ?? 0
+            let completedTxs: [CompletedTransaction] = (wallet.completedTransactions.0?.list.0 ?? [])
+            completedTxs.forEach { (tx) in
+                if tx.direction == .inbound {
+                    incomingTxs += 1
+                }
+            }
+
+            let balance = wallet.totalMicroTari.0 ?? MicroTari(0)
+
+            let triggers = type.triggers
+
+            guard incomingTxs >= triggers.numberOfIncomingTransactions &&
+            balance.rawValue >= triggers.totalBalance.rawValue &&
+                ICloudBackup.shared.isValidBackupExists() == triggers.hasConnectediCloud &&
+                (ICloudBackup.shared.lastBackup?.isEncrypted ?? false) == triggers.backupIsEncrypted else {
+                continue
+            }
+
+            UserDefaults.standard.set(true, forKey: type.userDefaultsKey)
+            let content = type.content
+            UserFeedback.shared.callToAction(
+                title: content.title,
+                boldedTitle: content.boldTitle,
+                description: content.description,
+                actionTitle: content.ctaButton,
+                cancelTitle: content.cancelButton,
+                onAction: {
+                    let profileVC = ProfileViewController()
+                    profileVC.goStraightToBackup = true
+                    vc.navigationController?.pushViewController(profileVC, animated: true)
```

Comment:
i don't really like solution to have goStraightToBackup property into profile VC. And all things that related to this property. I think we can have navigation controller with correct stack of VC ([ProfileVC, SettingsVC, BackupVC]) and then just push it.

navigationController.viewControllers = [VC,VC,VC]
i suppose it should help